ocp_resources/datavolume.py

```diff
@@ -216,20 +216,85 @@ def scratch_pvc(self):
             client=self.client,
         )
 
+    def _dv_phase(self):
+        if self.exists:
+            return self.instance.status.phase
+        return DV_DELETED_BY_GC
+
     def _check_none_pending_status(self, failure_timeout=120):
         # Avoid waiting for "Succeeded" status if DV's in Pending/None status
         sample = None
         try:
             for sample in TimeoutSampler(
                 wait_timeout=failure_timeout,
                 sleep=10,
-                func=lambda: self.instance.status.phase,
+                func=self._dv_phase,
             ):
                 # If DV status is Pending (or Status is not yet updated) continue to wait, else exit the wait loop
                 if sample in [self.Status.PENDING, None]:
                     continue
-                else:
-                    break
+                break
         except TimeoutExpiredError:
-            LOGGER.error(f"{self.name} status is {sample}")
+            self.logger.error(f"{self.name} status is {sample}")
             raise
+
+    def is_garbage_collector_enabled_dv(self):
+        if self.exists:
+            dv_annotations = self.instance.metadata.get("annotations")
+            if dv_annotations:
+                return (
+                    dv_annotations.get(
+                        f"{Resource.ApiGroup.CDI_KUBEVIRT_IO}/storage.deleteAfterCompletion"
+                    )
+                    is not False
+                )
+            return True
```

Comment:
if dv exists and there's no annotation - how does this assure that GC is enabled?

Reply:
please add a comment in the code